### PR TITLE
feat: move editor insert logic to `useSpecialContent` composable

### PR DESCRIPTION
**Pull request by scmmishra**
This PR makes the following changes

1. Move `createState` to a `editorHelpers`, we didn't need it inside the component anyway
2. Remove `insertMentionNode`, `insertCannedResponse` and `insertVariable` and replace with with a composable with abstracted insertion logic
3. Add `useSpecialContent` composable along with tests for it

## Motivation

The editor is one of the most complicated UI components we have, however it is too big of a component to be maintained meaningfully, and is largely untested. Refactoring it piece by piece will minimise the risk as well as get it ready for Vue 3 migration

### CODE REVIEW

**Comment by scmmishra** [comment]
@chatwoot/frontend hey folks, need your review on this. PTAL and let me know if there are any avenues for improvement, the plan is to rewrite this component completely in 4 to 5 iterations and ensure good tests

**Comment by scmmishra** [comment]
@pranavrajs PTAL

**Comment by scmmishra** [comment]
@pranavrajs @muhsin-k need your review on this

File: app/javascript/dashboard/composables/editor/useSpecialContent.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+   * @param {string} content - The canned response content
+   * @param {number} from - The start position of the canned response in the document
+   * @param {number} to - The end position of the canned response in the document
+   * @param {Object} variables - The variables to replace in the canned response
+   * @returns {Object} - The created canned response node and the updated from and to positions
+   */
+  const getCannedResponseNode = (editorView, content, from, to, variables) => {
+    const updatedMessage = replaceVariablesInMessage({
+      message: content,
+      variables: variables,
```
Comment (muhsin-k) [inline app/javascript/dashboard/composables/editor/useSpecialContent.js:42]:
```suggestion
      variables,
```

**Review by muhsin-k — APPROVED** [review]
The code LGTM and tested locally.

**Pull request closed without merging**